cowprotocol/user_segmentation/user_segmentation_4872780.sql

```diff
@@ -0,0 +1,122 @@
+-- Query that calculates for each [user, cal. month] pair a user status that is one of ['Loyal', 'Opportunistic', 'Lost', 'Gone', 'Didnt exist']
+-- Then it aggregates it by month and calculates transitions from one status to another month over month.
+-- It is parametrised by chain and project.
+-- Status descriptions:
+-- Loyal - only traded {{project}} during calendar month
+-- Opportunistic - traded both {{project}} during calendar month and other projects
+-- Gone - was created before date of interest but didn't trade in given calendar month
+-- Didnt exist - user was created this month (this status is only possible for prev_status column)
+--
+-- Parameters:
+--  {{start_time}} - the trade timestamp for which the analysis should start (inclusive)
+--  {{chain_name}} - chain to do the analysis for
+--  {{project}} - project to do the analysis for
+--  {{current_status}} - filter to limit analysis to users that had this status as current for given calendar month
+--  {{previous_status}} - filter to limit analysis to users that had this status as previous for given calendar month
+
+
+with creation_month_per_user as (
+    -- TODO: use both dex and dex-agg here
+    select
+        taker as user_address,
+        min(date_trunc('Month', block_date)) as creation_month
+    from
+        dex_aggregator.trades
+    where
+        blockchain = '{{chain_name}}'
+    group by 1
+),
+
+all_active_users as (
+    -- TODO: use layered txs here to prevent miss-information
+    select distinct
+        taker as user_address,
+        date_trunc('Month', block_date) as block_month,
+        case when project in ('1inch', '1inch Limit Order Protocol') then '1inch' else project end as project
+    from
+        dex_aggregator.trades
+    where
+        blockchain = '{{chain_name}}'
+        and
+        block_date >= timestamp '{{start_time}}' - interval '1' month
+),
+
+users_backbone as (
```

Comment:
maybe a small inline comment why this is needed (not super clear from the name and the sequence is not trivial to understand)